cpp/src/gandiva/precompiled/extended_math_ops.cc

```diff
@@ -249,6 +265,26 @@ ROUND_DECIMAL(float64)
 ROUND_DECIMAL_TO_SCALE(float32)
 ROUND_DECIMAL_TO_SCALE(float64)
 
+// rounds the number to the given scale
+FORCE_INLINE
+gdv_float64 bround_float64_int32(gdv_float64 num, gdv_int32 out_scale) {
+  if (out_scale < 0) {
+    out_scale = out_scale * (-1);
+  }
+  gdv_float64 scale_multiplier = get_scale_multiplier(out_scale);
+  gdv_float64 to_round = num * scale_multiplier;
```

Comment:
This can cause an overflow. This implementation has to be thought through - please do it in a separate PR

Reply:
Ok, I will open a new PR when as soon as this one is merged